src/use-observable.ts

```diff
@@ -2,42 +2,52 @@ import { Observable, BehaviorSubject } from 'rxjs'
 import { useState, useEffect, useMemo } from 'react'
 
 export type InputFactory<T, U = undefined> = U extends undefined
-  ? () => Observable<T>
-  : (props$: Observable<U>) => Observable<T>
+  ? (state$: Observable<T>) => Observable<T>
+  : (props$: Observable<U>, state$: Observable<T>) => Observable<T>
```

Comment:
I suggest to use `inputs$` instead of `props$` everywhere, cause we could pass other states from `useState` hooks into `useObservable`.